umqtt.simple/umqtt/simple.py

```diff
@@ -43,9 +45,15 @@ def connect(self, clean_session=True):
         msg = bytearray(b"\x10\0\0\x04MQTT\x04\x02\0\0")
         msg[1] = 10 + 2 + len(self.client_id)
         msg[9] = clean_session << 1
+        if self.user and self.pswd:
```

Comment:
You are right, if have well read the documentation, the user name and password can be empty, so I must check with ```if self.user is not None and self.pswd is not None```

But I don't understand the "optimised" version, because I can't "_send_str" None.

I have 3 choice:
  1. test ```if self.user is not None and self.pswd is not None```
  2. init with empty bytes: ```def __init__(self, client_id, server, port=0, ssl=False, user=None, password=b''):``` and test ```if self.user is not None```
  3. make separate action for user name and password
``` python
if self.user is not None:
    msg[1] += 2 + len(self.user)
    msg[9] |= 0x80
if self.pswd is not None:
    msg[1] +=  2 + len(self.pswd)
    msg[9] |= 0x80
```
...
``` python
if self.user is not None:
    self._send_str(self.user)
if self.pswd is not None:
    self._send_str(self.user)
```

Which choice is the best for you ?